test/node/src/safe-empty-where-in.test.ts

```diff
@@ -0,0 +1,916 @@
+        const query = db
+          .selectFrom('safeEmptyArrayPerson')
+          .where('firstName', 'in', ['John', 'Mary'])
+          .select('safeEmptyArrayPerson.firstName')
+
+        testSql(query, dialect, {
+          postgres: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in ($1, $2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mysql: {
+            sql: [
+              'select `safeEmptyArrayPerson`.`firstName`',
+              'from `safeEmptyArrayPerson`',
+              'where `firstName` in (?, ?)',
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mssql: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (@1, @2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (?, ?)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+        })
+
+        let result = await query.execute()
+
+        expect(result).to.have.length(2)
+        expect(result).to.deep.equal([
+          { firstName: 'John' },
+          { firstName: 'Mary' },
+        ])
+
+        const notInQuery = db
+          .selectFrom('safeEmptyArrayPerson')
+          .where('firstName', 'not in', ['John', 'Mary'])
+          .select('safeEmptyArrayPerson.firstName')
+
+        testSql(notInQuery, dialect, {
+          postgres: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in ($1, $2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mysql: {
+            sql: [
+              'select `safeEmptyArrayPerson`.`firstName`',
+              'from `safeEmptyArrayPerson`',
+              'where `firstName` not in (?, ?)',
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          mssql: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (@1, @2)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (?, ?)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+        })
+
+        result = await notInQuery.execute()
+
+        expect(result).to.have.length(1)
+        expect(result).to.deep.equal([{ firstName: 'Tom' }])
```

Comment:
make the first query have both in and not in to save time.

Reply:
this actually surfaced some bugs with this plugin. The AST for the case where we have `AND` or `OR` operators will not result in the transformation being applied. 

I could either:
- Apply the transformation on any `BinaryOperationNode`
- Repeat the transformation for WHERE, AND and OR nodes.

which should be the preferred option? Im afraid doing the transformation on any `BinaryOperationNode` would result in other side-effects

Reply:
I think applying this plugin for all `in` usage is the right way to go, because `in ()` is a universal problem for some people, not just `where ... in`.

Reply:
Great, resolving :)